### PR TITLE
gh-2557 Pull data from contracts, part 1

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Handles #2557

Changes proposed in this pull request:
- Refactor eth_call to a separate RpcClient class
- Get SafeToken.paused
- Get Airdrop.Vesting by vesting hash (vesting id)
- Get DelegateRegistry.delegate by delegator address

### CODE REVIEW

**Comment by biafra23** [comment]
I would prefer not to have spaces in file or directory names. But apparently that ship has sailed. But do we need to have a trailing space in a directory name like: [Multisig/Data/Services/Safe Claiming Service /GuardiansRequest.swift](https://github.com/safe-global/safe-ios/pull/2574/files#diff-2dd592adf754cc4891e0f96711e2c92db5f199a3f5e83f9ebbfda745ae86c776)

File: Multisig/Logic/SafeTransactionController.swift [hunk 1]
```diff
@@ -123,56 +123,17 @@ class SafeTransactionController {
...
-                let addresses = returns._arg0.elements.compactMap(Address.init)
-                dispatchOnMainThread(completion(.success(addresses)))
-            } catch {
-                dispatchOnMainThread(completion(.failure(error)))
+            input: GnosisSafe_v1_3_0.getOwners()
+        ) { contractResult in
+            let result = contractResult.map { returns in
+                returns._arg0.elements.compactMap(Address.init)
             }
+            dispatchOnMainThread(completion(result))
```
Comment (MouazAlzahabi) [inline Multisig/Logic/SafeTransactionController.swift]: We have the same line in RPC client send method

**Review by MouazAlzahabi — APPROVED** [review]

**Comment by DmitryBespalov** [comment]
>I would prefer not to have spaces in file or directory names. But apparently that ship has sailed. But do we need to have a trailing space in a directory name like: [Multisig/Data/Services/Safe Claiming Service /GuardiansRequest.swift](https://github.com/safe-global/safe-ios/pull/2574/files#diff-2dd592adf754cc4891e0f96711e2c92db5f199a3f5e83f9ebbfda745ae86c776)

Oh, I didn't notice that it was like that. I was using the existing folder structure. Will change in the next PR

File: Multisig/Logic/SafeTransactionController.swift (hunk 1 above)
Comment (DmitryBespalov) [reply Multisig/Logic/SafeTransactionController.swift]: yeah, true that it's enough if it's just in one place already

**Pull request merged**